doc/Operation.md

```diff
@@ -0,0 +1,328 @@
+# Operation\<T>
+
+The Azure SDK for C++ defines an Operation as an abstract class to support a long running operation ([LRO](http://restalk-patterns.org/long-running-operation-polling.html)).
+
+The SDK clients would define a concrete class from the `Operation<T>` which an end users can consume for getting the status of the running operation.
+
+## Operation Status
+
+The status of the operation defines the stages which the operation can be set from its creation and up to completion. The next table show the supported status of an `Operation<T>`.
+
+| Status     | Description                                                                    |
+| ---------- | ------------------------------------------------------------------------------ |
+| NotStarted | The operation is created but the server has not yet started it.                |
+| Running    | The operation is running on the server side and has not yet be completed.      |
+| Succeeded  | The server has successfully finished running the LRO. This is a `final stage`. |
+| Cancelled  | The operation was interrupted before completion. This is a `final stage`.      |
+| Failed     | The server failed to run the LRO. This is a `final stage`.                     |
+
+> Note: A final stage means that the status won't change anymore.
```

Comment:
Nice!

Reply:
@JeffreyRichter how would a service introduce more states if the OperationStatus is defined in Core?  Would they update the Core type for everybody?

Reply:
The property is just a string. The service would return some string.